modules/RsIskraMeter/src/main.rs

```diff
@@ -0,0 +1,1034 @@
+        self.write_single_register(7051, 0x4200)?;
+        log::info!("Started transaction.");
+
+        // Wait until the meter is active.
+        retry::retry(
+            || {
+                let state = self.read_state()?;
+                match state {
+                    IskraMaterState::Active => Ok(retry::RetryState::Done),
+                    _ => Ok(retry::RetryState::KeepTrying),
+                }
+            },
+            std::time::Duration::from_secs(1),
+        )?;
+
+        self.check_signature_status()?;
+        let signature = self.read_signature()?;
+        let signed_meter_values = self.read_signed_meter_values()?;
+        Ok(TransactionStartResponse {
+            error: Option::None,
+            signed_meter_value: Some(SignedMeterValue {
+                signed_meter_data: create_ocmf(signed_meter_values, signature),
+                signing_method: "OCMF".to_string(),
+                encoding_method: String::new(),
+                public_key: None,
+                timestamp: None,
+            }),
+            transaction_min_stop_time: Option::None,
+            status: TransactionRequestStatus::OK,
+            transaction_max_stop_time: Option::None,
+        })
+    }
+
+    fn stop_transaction(&self) -> Result<generated::types::powermeter::TransactionStopResponse> {
+        // We can start transaction only in Idle state
+        let state = self.read_state()?;
+        match state {
+            IskraMaterState::Idle => {
+                log::error!("The state of meter is Idle and we can not stop transaction",);
+                anyhow::bail!("Transaction not started");
+            }
+            IskraMaterState::Active
+            | IskraMaterState::Active_after_power_failure
+            | IskraMaterState::Active_after_reset => {}
+            IskraMaterState::Unknown => {
+                log::warn!("Unknown state")
+            }
+        }
+
+        // The state for start transaction is incorrect
+        self.write_single_register(7051, 0x7200)?;
+
+        // Wait until the meter is idle.
+        retry::retry(
+            || {
+                let state = self.read_state()?;
+                match state {
+                    IskraMaterState::Idle => Ok(retry::RetryState::Done),
+                    _ => Ok(retry::RetryState::KeepTrying),
+                }
+            },
+            std::time::Duration::from_secs(1),
+        )?;
+
+        self.check_signature_status()?;
+        let signature = self.read_signature()?;
+        let signed_meter_values = self.read_signed_meter_values()?;
+        Ok(TransactionStopResponse {
+            error: Option::None,
+            signed_meter_value: Some(SignedMeterValue {
+                signed_meter_data: create_ocmf(signed_meter_values, signature),
+                signing_method: "OCMF".to_string(),
+                encoding_method: String::new(),
+                public_key: None,
```

Comment:
If possible you can include the public key that's published in ready() here as well